host/hackrf-tools/src/hackrf_transfer.c

```diff
@@ -1387,6 +1388,10 @@ int main(int argc, char** argv)
 	interval_timer.it_value.tv_sec = 0;
 	setitimer(ITIMER_REAL, &interval_timer, NULL);
 #endif
+	if (transmit) {
+		// Wait for TX to finish.
+		hackrf_await_tx_flush(device);
```

Comment:
This can wait forever (see my previous PR comment) because `hackrf_stop_tx()` hasn't been called yet. I think the right place for this is where "hackrf_stop_tx() done\n" is printed. I would like flushing to happen in `signalsource` mode in addtion to `transmit` mode (for the case where `-n` is used with `-c`).

Reply:
That `hackrf_stop_tx()` hasn't been called yet is exactly the intent. We want to delay that call until the buffer is flushed, because once we make that call the device stops TX immediately.

The bug you found is that I didn't handle the case where the main loop is interrupted by Ctrl-C, rather than by the TX callback calling `stop_main_loop` before returning nonzero. In that scenario we shouldn't wait for a flush at all - the user wants to stop, so we should just call `hackrf_stop_tx` immediately.

Reply:
Fixed in new commits.